PhysicsTools/NanoAOD/python/triggerObjects_cff.py

```diff
@@ -218,19 +216,47 @@
         L1PreFiringWeight_Nom = ExtVar(cms.InputTag("prefiringweight:nonPrefiringProb"), "double", doc = "L1 pre-firing event correction weight (1-probability)", precision=8),
         L1PreFiringWeight_Up = ExtVar(cms.InputTag("prefiringweight:nonPrefiringProbUp"), "double", doc = "L1 pre-firing event correction weight (1-probability), up var.", precision=8),
         L1PreFiringWeight_Dn = ExtVar(cms.InputTag("prefiringweight:nonPrefiringProbDown"), "double", doc = "L1 pre-firing event correction weight (1-probability), down var.", precision=8),
+        L1PreFiringWeightMuon_Nom = ExtVar(cms.InputTag("prefiringweight:nonPrefiringProbMuon"), "double", doc = "Muon L1 pre-firing event correction weight (1-probability)", precision=8),
+        L1PreFiringWeightMuon_SystUp = ExtVar(cms.InputTag("prefiringweight:nonPrefiringProbMuonSystUp"), "double", doc = "Muon L1 pre-firing event correction weight (1-probability), up var. syst.", precision=8),
+        L1PreFiringWeightMuon_SystDn = ExtVar(cms.InputTag("prefiringweight:nonPrefiringProbMuonSystDown"), "double", doc = "Muon L1 pre-firing event correction weight (1-probability), down var. syst.", precision=8),
+        L1PreFiringWeightMuon_StatUp = ExtVar(cms.InputTag("prefiringweight:nonPrefiringProbMuonStatUp"), "double", doc = "Muon L1 pre-firing event correction weight (1-probability), up var. stat.", precision=8),
+        L1PreFiringWeightMuon_StatDn = ExtVar(cms.InputTag("prefiringweight:nonPrefiringProbMuonStatDown"), "double", doc = "Muon L1 pre-firing event correction weight (1-probability), down var. stat.", precision=8),
+        L1PreFiringWeightECAL_Nom = ExtVar(cms.InputTag("prefiringweight:nonPrefiringProbECAL"), "double", doc = "ECAL L1 pre-firing event correction weight (1-probability)", precision=8),
+        L1PreFiringWeightECAL_Up = ExtVar(cms.InputTag("prefiringweight:nonPrefiringProbECALUp"), "double", doc = "ECAL L1 pre-firing event correction weight (1-probability), up var.", precision=8),
+        L1PreFiringWeightECAL_Dn = ExtVar(cms.InputTag("prefiringweight:nonPrefiringProbECALDown"), "double", doc = "ECAL L1 pre-firing event correction weight (1-probability), down var.", precision=8),
     )
 )
 
 triggerObjectTables = cms.Sequence( unpackedPatTrigger + triggerObjectTable )
 
+#protect nano V8 production from changes to prefiring weights introduce for V9
+(run2_jme_2017 & run2_nanoAOD_106Xv1 & ~run2_nanoAOD_devel).toModify(prefiringweight,DataEraECAL = cms.string("2017BtoF"))
+(run2_nanoAOD_106Xv1 & ~run2_nanoAOD_devel).toModify(prefiringweight,DoMuons = cms.bool(False))
+(run2_nanoAOD_106Xv1 & ~run2_nanoAOD_devel).toModify(l1PreFiringEventWeightTable.variables , L1PreFiringWeightMuon_Nom = None, L1PreFiringWeightMuon_SystUp = None, L1PreFiringWeightMuon_SystDn = None, L1PreFiringWeightMuon_StatUp = None, L1PreFiringWeightMuon_StatDn = None, L1PreFiringWeightECAL_Nom = None, L1PreFiringWeightECAL_Up = None, L1PreFiringWeightECAL_Dn = None)
+
```

Comment:
@mariadalfonso Here we remove the new prefiring weights from the nano event content for V8, so having empty histograms for them in DQM is expected. What I am a puzzled about is why the prefiring weight is 1, it should still reflect the ECAL prefiring. What workflow did you use for that test?

Reply:
Mhhh, now that we have the UL maps for ECAL, this might need a further protection for that year. @lathomas